### ISSUE TITLE
compactIri does not choose most compact form

### ISSUE THREAD

**Issue by RubenVerborgh** [rubensworks/jsonld-context-parser.js#21]
With a context of

```json
{
  "@context": {
    "foaf": "http://xmlns.com/foaf/0.1/",
    "knows": "foaf:knows",
    "friends": "foaf:knows",
    "firstName": "foaf:givenName",
    "name": "http://xmlns.com/foaf/0.1/name",
    "label": "http://www.w3.org/2000/01/rdf-schema#label"
  }
}
```

`compactIri("http://xmlns.com/foaf/0.1/")` gives `foaf:knows` rather than `knows` or `friends`.

**Comment by rubensworks** [comment]
Right, it picks the first possible match in the context, independent of the type of compaction.

Note to self: add two levels of IRI compacting: prefixed terms, compact terms.
Compact terms should be preferred over prefixed terms.

**Comment by RobinDeBaets** [comment]
Another example of where the most compact form is not used:

```
let context = {
                "npmd": "https://linkedsoftwaredependencies.org/bundles/npm/",
                "chw": "npmd:@comunica/actor-init-hello-world/",
                "files-chw": "chw:^1.0.0/",
                "config-example": "files-chw:config/config-example.json#",

                "ActorInitHelloWorld": "chw:Actor/Init/HelloWorld",
                "hello": "chw:hello"
};
let fullPath = "npmd:@comunica/actor-init-hello-world/Actor/Init/HelloWorld";
console.log(ContextParser.compactIri(fullPath, context));
```
Results in `chw:Actor/Init/HelloWorld`, but `ActorInitHelloWorld` would have been shorter

**Comment by rubensworks** [comment]
Note to self: we'll need some looping.

**Comment by matthieu-fesselier** [comment]
it's not working as expected on my side. Am I doing something wrong?
```javascript
import JSONLDContextParser from 'https://dev.jspm.io/jsonld-context-parser@1.3.2';
     
const ContextParser = JSONLDContextParser.ContextParser;
const myParser = new ContextParser();
     
const base_context = {
  rdfs: 'http://www.w3.org/2000/01/rdf-schema#',
  name: 'rdfs:label'
};


(async () => {
  let context = await myParser.parse(base_context);
  let iri = ContextParser.compactIri("http://www.w3.org/2000/01/rdf-schema#label", context, true);
  console.log(iri); // prints "rdfs:label", I expected "name"
})();
```

JSBin here: https://jsbin.com/rajisadumo/edit?html,console

**Comment by rubensworks** [comment]
Your code looks correct to me, so this must be a bug.

**Comment by rubensworks** [comment]
Fixed in 1.3.4.
